File: Leetcode/3541 - Find Most Frequent Vowel and Consonant/solution.js
var maxFreqSum = function(s) {
    const isVowel = (c) => {
        return c === 'a' || c === 'e' || c === 'i' || c === 'o' || c === 'u';
    };

    const charFreq = new Array(26).fill(0);
    let maxVowelFreq = 0;
    let maxConsonantFreq = 0;

    for (const c of s) {
        const idx = c.charCodeAt(0) - 97;
        const freq = ++charFreq[idx];
        if (isVowel(c)) {
            maxVowelFreq = Math.max(maxVowelFreq, freq);
        } else {
            maxConsonantFreq = Math.max(maxConsonantFreq, freq);
        }
    }
    return maxVowelFreq + maxConsonantFreq;
};

File: Leetcode/3541 - Find Most Frequent Vowel and Consonant/solution.cpp
bool isVowel(char c) {
    return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
}

int maxFreqSum(string s) {
    vector<int> charFreq(26, 0);
    int maxVowelFreq = 0, maxConsonantFreq = 0;

    for (char c: s) {
        charFreq[c - 'a']++;
        if (isVowel(c)) {
            maxVowelFreq = max(maxVowelFreq, charFreq[c - 'a']);
        } else {
            maxConsonantFreq = max(maxConsonantFreq, charFreq[c - 'a']);
        }
    }

    return maxVowelFreq + maxConsonantFreq;
}